driver/mseloss_driver.hpp

```diff
@@ -0,0 +1,732 @@
+                                                divisor);
+
+            if(status != miopenStatusSuccess)
+            {
+                std::cerr << "Error: miopenMSELossBackward failed" << std::endl;
+                return status;
+            }
+
+            float time = 0.0;
+            miopenGetKernelTime(GetHandle(), &time);
+            kernel_total_time += time;
+            if(i == 0)
+                kernel_first_time = time;
+        }
+    }
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        STOP_TIME
+        int iter = inflags.GetValueInt("iter");
+        if(WALL_CLOCK)
+            std::cout << "Wall-clock Time Backward MSELoss Elapsed: " << t.gettime_ms() / iter
+                      << " ms\n";
+        float kernel_average_time = iter > 1 ? kernel_total_time / (iter - 1) : kernel_first_time;
+        std::cout << "Kernel Backward Time Elapsed: " << kernel_average_time << " ms\n";
+    }
+
+    if(input_grad_buf->FromGPU(GetStream(), input_grad.data()) != miopenStatusSuccess)
+        std::cerr << "Error: Failed to copy input_grad from GPU, size " << input_grad.size()
+                  << std::endl;
+    if(target_grad_buf->FromGPU(GetStream(), target_grad.data()) != miopenStatusSuccess)
+        std::cerr << "Error: Failed to copy target_grad from GPU, size " << target_grad.size()
+                  << std::endl;
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int MSELossDriver<Tgpu, Tref>::RunBackwardCPU()
+{
+    if(divisor == 0)
+    {
+        mloMSELossBackwardUnreducedRunHost(inputDesc,
+                                           targetDesc,
+                                           outputDesc,
+                                           inputGradDesc,
+                                           targetGradDesc,
+                                           input.data(),
+                                           target.data(),
+                                           output.data(),
+                                           input_grad_host.data(),
+                                           target_grad_host.data());
+    }
+    else
+    {
+        mloMSELossBackwardRunHost(inputDesc,
+                                  targetDesc,
+                                  outputDesc,
+                                  inputGradDesc,
+                                  targetGradDesc,
+                                  input.data(),
+                                  target.data(),
+                                  output_host.data(),
+                                  input_grad_host.data(),
+                                  target_grad_host.data(),
+                                  divisor);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int MSELossDriver<Tgpu, Tref>::VerifyBackward()
+{
+    RunBackwardCPU();
+
+    for(size_t i = 0; i < input_grad.size(); i++)
```

Comment:
Please use miopen::rms_range.
Or, if you have a reason for doing this, please tell me.

Reply:
It was associated with the precision issue. I’ve changed to `rms_range`